src/examples/mfem_common_plane.cpp

```diff
@@ -0,0 +1,429 @@
+    attrib_marker = 0;
+    for (auto move_attrib : move_attribs)
+    {
+      attrib_marker[move_attrib-1] = 1;
+    }
+    mfem::Array<int> vdof_marker(par_fe_space.GetVSize());
+    vdof_marker = 0;
+    for (int e{0}; e < pmesh->GetNE(); ++e)
+    {
+      if (attrib_marker[pmesh->GetElement(e)->GetAttribute()-1])
+      {
+        mfem::Array<int> vdofs;
+        par_fe_space.GetElementDofs(e, vdofs);
+        par_fe_space.DofsToVDofs(2, vdofs);
+        for (auto vdof : vdofs)
+        {
+          vdof_marker[vdof] = 1;
+        }
+      }
+    }
+    for (int i{0}; i < vdof_marker.Size(); ++i)
+    {
+      if (vdof_marker[i])
+      {
+        v[i] = -std::abs(initial_v);
+      }
+    }
+  }
+
+  // recover dirichlet bc tdof list
+  mfem::Array<int> ess_vdof_list;
+  {
+    mfem::Array<int> ess_vdof_marker;
+    mfem::Array<int> ess_bdr(pmesh->bdr_attributes.Max());
+    ess_bdr = 0;
+    for (auto fix_attrib : fix_attribs)
+    {
+      ess_bdr[fix_attrib-1] = 1;
+    }
+    par_fe_space.GetEssentialVDofs(ess_bdr, ess_vdof_marker);
+    mfem::FiniteElementSpace::MarkerToList(ess_vdof_marker, ess_vdof_list);
+  }
+  timer.stop();
+  SLIC_INFO_ROOT(axom::fmt::format(
+    "Time to set up boundary conditions: {0:f}ms", timer.elapsedTimeInMilliSec()
+  ));
+
+  // set up mfem elasticity bilinear form
+  timer.start();
+  mfem::ConstantCoefficient rho {100.0};
+  mfem::ConstantCoefficient lambda {100000.0};
+  mfem::ConstantCoefficient mu {100000.0};
+  ExplicitMechanics op {par_fe_space, rho, lambda, mu};
+  timer.stop();
+  SLIC_INFO_ROOT(axom::fmt::format(
+    "Time to set up elasticity bilinear form: {0:f}ms", timer.elapsedTimeInMilliSec()
+  ));
+
+  // set up time integrator
+  CentralDiffSolver solver { ess_vdof_list };
+  solver.Init(op);
+
+  // set up tribol
+  timer.start();
+  tribol::initialize(pmesh->SpaceDimension(), MPI_COMM_WORLD);
+  tribol::registerMfemMesh(
+    0, 0, 1, *pmesh, coords, surf1_attribs, surf2_attribs,
+    tribol::SURFACE_TO_SURFACE, 
+    tribol::NO_CASE, 
+    tribol::COMMON_PLANE, 
+    tribol::FRICTIONLESS,
+    tribol::PENALTY,
+    tribol::BINNING_GRID
+  );
+  tribol::registerMfemVelocity(0, v);
+  tribol::setPenaltyOptions(
+    0,
+    tribol::KINEMATIC,
+    tribol::KINEMATIC_CONSTANT,
+    tribol::NO_RATE_PENALTY
+  );
+  tribol::setKinematicConstantPenalty(0, p_kine);
+  tribol::setKinematicConstantPenalty(1, p_kine);
+  timer.stop();
+  SLIC_INFO_ROOT(axom::fmt::format(
+    "Time to set up Tribol: {0:f}ms", timer.elapsedTimeInMilliSec()
+  ));
+
+  int cycle {0};
+  for (double t {0.0}; t < t_end; t+=dt)
+  {
+    dc_pv.SetCycle(cycle);
+    dc_pv.SetTime(t);
+    dc_pv.SetTimeStep(dt);
+    dc_vi.SetCycle(cycle);
+    dc_vi.SetTime(t);
+    dc_vi.SetTimeStep(dt);
+
+    if (cycle % output_cycles == 0)
+    {
+      dc_pv.Save();
+      dc_vi.Save();
+    }
+
+    tribol::update(cycle, t, dt);
+
+    op.SetTime(t);
+    solver.Step(u, v, t, dt);
+
+    coords += u;
+    if (order == 1)
```

Comment:
Do we not need to do this if order > 1?

Reply:
No, if `order > 1`, then nodal locations are stored in the `Nodes` grid function.  Comment has been added.